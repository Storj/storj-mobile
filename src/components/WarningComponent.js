import {
    View,
    Text,
    StyleSheet,
    StatusBar
} from 'react-native';
import React, { Component } from 'react';
import { getHeight, getWidth } from '../utils/adaptive';
import PropTypes from 'prop-types';

/**
 * Red warning on top of application with message 
 */
export default class WarningComponent extends Component{
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <View style = { this.props.message ? styles.secretPhraseButton : styles.secretPhraseHide } >
                <StatusBar backgroundColor = { this.props.statusBarColor }/>
                <Text style = { styles.secretPhraseText }>{this.props.message}</Text> 
            </View>
        );
    }
}

WarningComponent.propTypes = {
    message: PropTypes.string,
    statusBarColor: PropTypes.object
};

const styles = StyleSheet.create({
    secretPhraseText: { 
        fontFamily: 'montserrat_medium', 
        fontSize: getHeight(12), 
        color: '#FFFFFF' 
    },
    secretPhraseButton: { 
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        height: getHeight(20), 
        backgroundColor: '#EB5757', 
        justifyContent: 'space-between',
        paddingHorizontal: getWidth(20),
        alignItems: 'center', 
        flexDirection: 'row' 
    },
    secretPhraseHide: { 
        position: 'absolute',
        top: 0,
        left: 0,
        right: 0,
        height: getHeight(0), 
        backgroundColor: '#EB5757', 
        justifyContent: 'space-between',
        paddingHorizontal: getWidth(20),
        alignItems: 'center', 
        flexDirection: 'row' 
    },
    expandImage: { 
        height: getHeight(24), 
        width: getWidth(24)
    },
});
