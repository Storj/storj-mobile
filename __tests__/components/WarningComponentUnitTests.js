import {
    StatusBar,
    Text
} from 'react-native';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WarningComponent from '../../src/components/Common/WarningComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('WarningComponent', () => {

    it('renders correctly', () => {
        const wrapper = shallow(
            <WarningComponent />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with props', () => {
        const wrapper = shallow(
            <WarningComponent 
                message = { 'test message' }
                statusBarColor = { 'red' } />
        );

        expect(wrapper.containsMatchingElement(<StatusBar backgroundColor = { 'red' }/>)).toBe(true);
        expect(wrapper.containsMatchingElement(<Text>test message</Text>)).toBe(true);
    });
});