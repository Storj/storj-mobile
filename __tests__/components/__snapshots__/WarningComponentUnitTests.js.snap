// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningComponent renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WarningComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          backgroundColor={undefined}
          showHideTransition="fade"
        />,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 24,
            }
          }
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#EB5757",
        "flexDirection": "row",
        "height": 0,
        "justifyContent": "space-between",
        "left": 0,
        "paddingHorizontal": 40,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "backgroundColor": undefined,
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Object {
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 24,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            backgroundColor={undefined}
            showHideTransition="fade"
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 24,
              }
            }
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#EB5757",
          "flexDirection": "row",
          "height": 0,
          "justifyContent": "space-between",
          "left": 0,
          "paddingHorizontal": 40,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "backgroundColor": undefined,
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 24,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
